src/Tools/dotnet-monitor/Profiler/ProfilerService.cs

```diff
@@ -52,18 +54,63 @@ public async Task ApplyProfiler(IEndpointInfo endpointInfo, CancellationToken ca
                 DiagnosticsClient client = new DiagnosticsClient(endpointInfo.Endpoint);
                 Dictionary<string, string> env = await client.GetProcessEnvironmentAsync(cancellationToken);
 
-                // CONSIDER: Allow alternate means of specifying or heuristically determining the runtime identifier
-                // - For Windows and OSX, build identifier from platform + architecture of target process.
-                // - Lookup same environment variable on dotnet-monitor itself.
-                // - Check libc type of dotnet-monitor.
-                env.TryGetValue(ToolIdentifiers.EnvironmentVariables.RuntimeIdentifier, out string runtimeIdentifier);
+                if (!env.TryGetValue(ToolIdentifiers.EnvironmentVariables.RuntimeIdentifier, out string runtimeIdentifier))
+                {
+                    ProcessInfo processInfo = await client.GetProcessInfoAsync(cancellationToken);
+
+                    // This is mostly correct, except that "arm" and "armv6" are both reported as "arm32".
+                    string ridArchitecture = processInfo.ProcessArchitecture;
+                    Debug.Assert(!"arm32".Equals(ridArchitecture, StringComparison.Ordinal), "Unable to distinguish arm from armv6");
+
+                    if (!string.IsNullOrEmpty(ridArchitecture) && !ridArchitecture.Equals("Unknown", StringComparison.Ordinal))
+                    {
+                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        {
+                            runtimeIdentifier = FormattableString.Invariant($"win-{ridArchitecture}");
+                        }
+                        else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                        {
+                            runtimeIdentifier = FormattableString.Invariant($"osx-{ridArchitecture}");
+                        }
+                        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                        {
+                            // Make best effort to determine which type of C library is used
+                            // in order to pick the correct Linux RID.
+                            try
+                            {
+                                Process process = Process.GetProcessById((int)processInfo.ProcessId);
```

Comment:
It's my understanding that it just fails to load the profiler and doesn't affect the app.

This is a best effort scenario since we couldn't get the exact portable rid (which is NYI) nor was it told to us exactly.